### DISCUSSION TITLE
"Invalid non-printable ASCII character in URL" since 0.24.0

### DISCUSSION THREAD

**Discussion by vphpersson** [encode/httpx#2676] [Potential Issue] [answered]
Starting from version 0.24.0, I am no longer able to use `\x1f` in URLs. The exception I'm now seeing is raised from here:

```python
# If a URL includes any ASCII control characters including \t, \r, \n,
# then treat it as invalid.
if any(char.isascii() and not char.isprintable() for char in url):
    raise InvalidURL("Invalid non-printable ASCII character in URL")
```

Is there a particular reason for not allowing non-printable ASCII characters anymore?

**Comment by lovelydinosaur** [comment, marked as answer]
https://www.rfc-editor.org/rfc/rfc3986#section-3.3

**Reply by vphpersson** [reply]
I see. I haven't looked at the code for the previous version, but I assume characters such as `\x1f` were percent-encoded if provided. With the current version, I now have to do that myself?

**Reply by lovelydinosaur** [reply]
tested your assumption and yep.
